pxtcompiler/emitter/decompiler.ts

```diff
@@ -2821,49 +2851,56 @@ ${output}</xml>`;
         }
 
         function checkNamespaceDeclaration(n: ts.NamespaceDeclaration) {
-            if (!isModuleBlock(n.body)) {
-                return Util.lf("Namespaces cannot be nested.")
-            }
+            const kindCheck = checkKindNamespaceDeclaration(n, env);
+            if (!kindCheck) return undefined;
 
-            const kindInfo = env.blocks.kindsByName[n.name.text]
-            if (!kindInfo) {
-                return Util.lf("Only namespaces with 'kind' blocks can be decompiled");
-            }
+            const tilesetCheck = checkTilesetNamespace(n);
+            if (!tilesetCheck) return undefined;
 
-            const fail = Util.lf("Namespaces may only contain valid 'kind' exports");
+            return kindCheck || tilesetCheck;
```

Comment:
This just always returns kindCheck at this point, right? if `!kindCheck` it returned undefined earlier, so the `|| tilesetCheck` should always be short circuited? I suppose it doesn't matter as either way it's just returning an error on why this one will be a gray block that isn't commonly seen

Reply:
Yep, I'll remove the `||`. Users never actually see these errors anyways